pennylane/qnode.py

```diff
@@ -546,6 +549,14 @@ def __call__(self, *args, **kwargs):
         # construct the tape
         self.construct(args, kwargs)
 
+        cache = self.execute_kwargs.get("cache", False)
+        using_custom_cache = (
+            hasattr(cache, "__getitem__")
+            and hasattr(cache, "__setitem__")
+            and hasattr(cache, "__delitem__")
+        )
+        self._tape_cached = using_custom_cache and self.tape.hash in cache

```

Comment:
@antalszava I wonder if it would be a better pattern to do?
```suggestion
        cache = self.execute_kwargs.get("cache", False)
        try:
            self._tape_cached = self.tape.hash in cache
        except TypeError:
            self._tape_cached = False
```

This is minor though! Your call.

Reply:
I'd rather keep it, as the docstring mentions that `__getitem__`, `__setitem__` and `__delitem__` are required to be implemented for a cache.